applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -1937,6 +1989,220 @@ void TwoFluidNavierStokes<TElementData>::ComputeSplitting(
+            }
+        }
+    }
+
+    noalias(rRHS) += rhs;
+}
+
+template <class TElementData>
+void TwoFluidNavierStokes<TElementData>::PressureGradientStabilization(
+        TElementData& rData,
+        const Kratos::Vector& rInterfaceWeights,
+        const Matrix& rEnrInterfaceShapeFunctionPos,
+        const Matrix& rEnrInterfaceShapeFunctionNeg,
+        const GeometryType::ShapeFunctionsGradientsType& rInterfaceShapeDerivatives,
+        MatrixType& rKeeTot,
+		VectorType& rRHSeeTot)
+{
+    MatrixType kee = ZeroMatrix(NumNodes, NumNodes);
+    VectorType rhs_enr = ZeroVector(NumNodes);
+
+    Matrix enr_neg_interp = ZeroMatrix(NumNodes, NumNodes);
+    Matrix enr_pos_interp = ZeroMatrix(NumNodes, NumNodes);
+
+    double positive_density = 0.0;
+    double negative_density = 0.0;
+    double positive_viscosity = 0.0;
+    double negative_viscosity = 0.0;
+
+    for (unsigned int i = 0; i < NumNodes; i++){
+        if (rData.Distance[i] > 0.0){
+            enr_neg_interp(i, i) = 1.0;
+            positive_density = rData.NodalDensity[i];
+            positive_viscosity = rData.NodalDynamicViscosity[i];
+        } else{
+            enr_pos_interp(i, i) = 1.0;
+            negative_density = rData.NodalDensity[i];
+            negative_viscosity = rData.NodalDynamicViscosity[i];
+        }
+    }
+
+    GeometryType::ShapeFunctionsGradientsType EnrichedInterfaceShapeDerivativesPos = rInterfaceShapeDerivatives;
+    GeometryType::ShapeFunctionsGradientsType EnrichedInterfaceShapeDerivativesNeg = rInterfaceShapeDerivatives;
+
+    for (unsigned int i = 0; i < rInterfaceShapeDerivatives.size(); i++){
+        EnrichedInterfaceShapeDerivativesPos[i] = prod(enr_pos_interp, rInterfaceShapeDerivatives[i]);
+    }
+
+    for (unsigned int i = 0; i < rInterfaceShapeDerivatives.size(); i++){
+        EnrichedInterfaceShapeDerivativesNeg[i] = prod(enr_neg_interp, rInterfaceShapeDerivatives[i]);
+    }
+
+    double positive_volume = 0.0;
+    double negative_volume = 0.0;
+    for (unsigned int igauss_pos = 0; igauss_pos < rData.w_gauss_pos_side.size(); ++igauss_pos){
+        positive_volume += rData.w_gauss_pos_side[igauss_pos];
+    }
+
+    for (unsigned int igauss_neg = 0; igauss_neg < rData.w_gauss_neg_side.size(); ++igauss_neg){
+        negative_volume += rData.w_gauss_neg_side[igauss_neg];
+    }
+    const double element_volume = positive_volume + negative_volume;
+
+    GeometryType::Pointer p_geom = this->pGetGeometry();
+    const double h_elem = ElementSizeCalculator<Dim,NumNodes>::AverageElementSize(*p_geom);
+
+    double cut_area = 0.0;
+    for (unsigned int gp = 0; gp < rInterfaceWeights.size(); gp++){
+        cut_area += rInterfaceWeights[gp];
+    }
+
+    const double density = 1.0/(1.0/positive_density + 1.0/negative_density);
+    const double viscosity = 1.0/(1.0/positive_viscosity + 1.0/negative_viscosity);
+
+    // Stabilization parameters
+    const double coefficient = 1.0e0;
```

Comment:
I would use 1.0, Is there a specific reason for this?

Reply:
I will change it to 1.0. It is from my old checks (1.0e1, etc.).